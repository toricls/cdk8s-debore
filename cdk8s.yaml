imports:
  - k8s@1.16.0
  # In addition to the above, this example has imported "autoscaling/v2beta2/HorizontalPodAutoscaler" by executing `cdk8s import k8s --language typescript --include io.k8s.api.autoscaling.v2beta2.HorizontalPodAutoscaler`
