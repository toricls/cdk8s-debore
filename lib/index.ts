/**
 * Hello class.
 */
export class Hello {
  /**
   * Hey there!
   */
  public world() {
    return 'Hello, world!';
  }
}
