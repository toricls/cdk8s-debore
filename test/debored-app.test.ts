import { Testing, Chart } from 'cdk8s';
import { DeboredApp, IngressType } from '../lib';

test('minimal configuration', () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  
  // WHEN
  new DeboredApp(chart, 'myapp', {
    image: 'myimage',
  });

  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});

test('"namespace" controls the k8s namespace to use"', () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  
  // WHEN
  new DeboredApp(chart, 'myapp', {
    image: 'myimage',
    namespace: 'hello.namespace',
  });

  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});

test('"autoScale" will define pod autoscaling with sensible configuration', () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  
  // WHEN
  new DeboredApp(chart, 'myapp', {
    image: 'myimage',
    autoScale: true,
  });

  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});

test('"defaultReplicas" can be used to control the number of replicas', () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  
  // WHEN
  new DeboredApp(chart, 'myapp', {
    image: 'myimage',
    defaultReplicas: 10,
  });

  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});

test('"ingress == CLUSTER_IP" uses a ClusterIP service', () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  
  // WHEN
  new DeboredApp(chart, 'myapp', {
    image: 'myimage',
    ingress: IngressType.CLUSTER_IP,
  });

  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});

test('"ingress == NGINX_INGRESS" uses an nginx ingress', () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  
  // WHEN
  new DeboredApp(chart, 'myapp', {
    image: 'myimage',
    ingress: IngressType.NGINX_INGRESS,
  });

  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});
