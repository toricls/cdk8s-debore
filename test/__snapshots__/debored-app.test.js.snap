// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"autoScale" will define pod autoscaling with sensible configuration 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-myapp-deployment-deployment-c1cb76c8",
      "namespace": "default",
    },
    "spec": Object {
      "selector": Object {
        "matchLabels": Object {
          "app": "testmyappD19744AC",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "testmyappD19744AC",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "myimage",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "autoscaling/v2beta2",
    "kind": "HorizontalPodAutoscaler",
    "metadata": Object {
      "name": "test-myapp-deployment-hpa-a67148ea",
      "namespace": "default",
    },
    "spec": Object {
      "maxReplicas": 10,
      "metrics": Array [
        Object {
          "resource": Object {
            "name": "cpu",
            "target": Object {
              "averageUtilization": 85,
              "type": "Utilization",
            },
          },
          "type": "Resource",
        },
        Object {
          "resource": Object {
            "name": "memory",
            "target": Object {
              "averageUtilization": 75,
              "type": "Utilization",
            },
          },
          "type": "Resource",
        },
      ],
      "minReplicas": 1,
      "scaleTargetRef": Object {
        "apiVersion": "apps/v1",
        "kind": "Deployment",
        "name": "test-myapp-deployment-deployment-c1cb76c8",
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-myapp-exposable-service-568386d8",
      "namespace": "default",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "testmyappD19744AC",
      },
      "type": "ClusterIP",
    },
  },
]
`;

exports[`"ingress == NGINX_INGRESS" uses an nginx ingress 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-myapp-deployment-deployment-c1cb76c8",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "testmyappD19744AC",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "testmyappD19744AC",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "myimage",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-myapp-exposable-service-568386d8",
      "namespace": "default",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "testmyappD19744AC",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "annotations": Object {
        "kubernetes.io/ingress.class": "nginx",
        "nginx.ingress.kubernetes.io/rewrite-target": "/",
      },
      "name": "test-myapp-exposable-ingress-3f9c21c2",
      "namespace": "default",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "test-myapp-exposable-service-568386d8",
                  "servicePort": 80,
                },
                "path": "/test-myapp-deployment-deployment-c1cb76c8",
              },
            ],
          },
        },
      ],
    },
  },
]
`;

exports[`"ingress == SERVICE_LOADBALANCER" uses a LoadBalancer service 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-myapp-deployment-deployment-c1cb76c8",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "testmyappD19744AC",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "testmyappD19744AC",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "myimage",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-myapp-exposable-service-568386d8",
      "namespace": "default",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "testmyappD19744AC",
      },
      "type": "LoadBalancer",
    },
  },
]
`;

exports[`"namespace" controls the k8s namespace to use" 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-myapp-deployment-deployment-c1cb76c8",
      "namespace": "hello.namespace",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "testmyappD19744AC",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "testmyappD19744AC",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "myimage",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-myapp-exposable-service-568386d8",
      "namespace": "hello.namespace",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "testmyappD19744AC",
      },
      "type": "ClusterIP",
    },
  },
]
`;

exports[`"replicas" can be used to control the number of replicas 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-myapp-deployment-deployment-c1cb76c8",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 10,
      "selector": Object {
        "matchLabels": Object {
          "app": "testmyappD19744AC",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "testmyappD19744AC",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "myimage",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-myapp-exposable-service-568386d8",
      "namespace": "default",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "testmyappD19744AC",
      },
      "type": "ClusterIP",
    },
  },
]
`;

exports[`minimal configuration 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-myapp-deployment-deployment-c1cb76c8",
      "namespace": "default",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "testmyappD19744AC",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "testmyappD19744AC",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "myimage",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-myapp-exposable-service-568386d8",
      "namespace": "default",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "testmyappD19744AC",
      },
      "type": "ClusterIP",
    },
  },
]
`;
